added initial aio chart

k1rk opened the pull request:
for now no VMCluster
also even for VMSingle not all cases are tested.

currently service monitors are in progress

more about status in readme

aeciopires reviewed (changes requested):
Hi, @k1rk!

Thank you very much for your contribution.

I made some comments in order to increase the quality of the chart and make the use and documentation easier.

Best regards

k1rk commented on charts/victoria-metrics-k8s-stack/templates/servicemonitors/grafana.yaml:
```diff
@@ -0,0 +1,23 @@
+      app.kubernetes.io/instance: {{ $.Release.Name | quote }}
+{{- if empty .Values.grafana.vmServiceScrape.spec }}
+  endpoints:
+  - port: {{ .Values.grafana.service.portName }}
```
this one is expected. as we include grafana chart as dependency, we have default values from it. we can override them inside our `values.yaml` we dont want to override all values

aeciopires replied:
Hi, @k1rk!
 
Understand. It makes sense.

To be intuitive and make it easier for new users, I suggest editing the file `` README.md.gotmpl`` and adding the following paragraph with the explanation of how to change the values that are in the helm charts declared as dependency. This way, when the command `` helm-docs`` is executed, this paragraph will also be automatically inserted in the file `` README.md``.

Paragraph example:

> ATTENTION !!! Some default values are inherited from helm charts declared as a dependency in the `` Chart.yaml`` file. If you want to change the default values that are inherited, access the documentation and the `` values.yaml`` file for each chart to understand the content and create a customized values.yaml file for your need.


What do you think?

k1rk replied:
good idea. Added notice to main readme template as well as comments inside `values.yaml`

k1rk commented on charts/victoria-metrics-k8s-stack/Chart.yaml:
```diff
@@ -0,0 +1,41 @@
+apiVersion: v2
+name: victoria-metrics-k8s-stack
+description: A Helm chart for Kubernetes
```
sure thing

k1rk replied:
cleaned up

k1rk commented:
just quick clarfification. work on this chart is now finished. its "initial version" push to split PRs for easier review

k1rk commented on charts/victoria-metrics-k8s-stack/README.md:
```diff
@@ -0,0 +1,23 @@
+### plans
```
added, thanks

aeciopires commented:
Hi @k1rk 

Again, thank you very much for your contribution and for applying the suggestions for improvement. Count on me to review Pull Resquests and to help increase the quality of your work.

k1rk commented:
@aeciopires Hi. Thank for your review. I addressed all the comments, also added VMRules

Now all of the created dashboards should be fully working as well as alerts

also for now this setup working only with one replica of vm-single. additional functionality i suggest to split to separate PRs. this one already to hard to review

aeciopires commented:
Thanks @k1rk!

@tenmozes / @f41gh7  can they complement the review?

I looked at all the files and I think the work is already of good quality for merge. But feel free to review, test and see if there is anything else to improve.

@k1rk agree with the strategy of adding new features in another PR to facilitate the review. This was the biggest PR I have ever reviewed in my life! Hahahaha

f41gh7 commented:
Thanks for PR!
 
Still testing it, have some troubles with grafana.

f41gh7 commented on charts/victoria-metrics-k8s-stack/README.md.gotmpl:
```diff
@@ -0,0 +1,138 @@
+
+* Install the follow packages: ``git``, ``kubectl``, ``helm``, ``helm-docs``. See this [tutorial](../../REQUIREMENTS.md).
+
+* PV support on underlying infrastructure.
```
Also, suggest to add following requirements:
```bash
 helm repo add grafana https://grafana.github.io/helm-charts
 helm repo add ksm https://kubernetes.github.io/kube-state-metrics 
 helm repo add prometheus-community https://prometheus-community.github.io/helm-charts 
 helm update
```

k1rk replied:
added

aeciopires replied:
Exists ``helm update`` command for update repo?

Wouldn't the correct ``helm repo update`` be backwards compatible?

f41gh7 replied:
Thanks, changed it for helm repo update

f41gh7 commented on charts/victoria-metrics-k8s-stack/values.yaml:
```diff
@@ -0,0 +1,449 @@
+
+## Component scraping the kube controller manager
+kubeControllerManager:
+  enabled: true
```
is it better to disable by default?

k1rk replied:
i would go with default "true" and adding note to docs that for managed kubernetes like EKS,GKE you need to disable it

f41gh7 commented on charts/victoria-metrics-k8s-stack/values.yaml:
```diff
@@ -0,0 +1,449 @@
+## Component scraping kube scheduler
+##
+kubeScheduler:
+  enabled: true
```
+ for disabling.

k1rk replied:
same here

f41gh7 commented on charts/victoria-metrics-k8s-stack/templates/_helpers.tpl:
```diff
@@ -0,0 +1,126 @@
+{{- $vmAlertStackRemoteWrite := dict "remoteWrite" ( include "victoria-metrics-k8s-stack.vmEndpoint" . | fromYaml ) -}}
+{{- $vmAlertStackRemoteRead := dict "remoteRead" ( include "victoria-metrics-k8s-stack.vmEndpoint" . | fromYaml ) -}}
+{{- $vmAlertStackDatasource := dict "datasource" ( include "victoria-metrics-k8s-stack.vmEndpoint" . | fromYaml ) -}}
+{{- $vmAlertStackNotifier := dict "notifier"  ( dict "url" ( printf "http://vmalertmanager-%s.%s.svc:9093" (include "victoria-metrics-k8s-stack.fullname" .) .Release.Namespace ) ) -}}
```
i suggest to use notifiers instead of notifier and build per pod notifier url according to replica size of alertmanager.

k1rk replied:
lets have this in separate PR, to be able also to test it separatelly

f41gh7 replied:
agreed

f41gh7 commented on charts/victoria-metrics-k8s-stack/templates/grafana/datasource.yaml:
```diff
@@ -0,0 +1,36 @@
+{{- range until (int .Values.vmsingle.spec.replicaCount) }}
+    - name: VictoriaMetrics-{{ . }}
+      type: prometheus
+      url: "http://{{ include "victoria-metrics-k8s-stack.vmsingleName" $ }}-{{ . }}.{{ $.Release.Namespace }}.svc:{{ $.Values.vmsingle.spec.port | default 8428 }}"
```
hm, 8429 instead of 8428?

k1rk replied:
yep. operator uses 8429 by default for VMSingle. i was also confused =)

f41gh7 replied:
lets use 8429 as default )

f41gh7 commented on charts/victoria-metrics-k8s-stack/values.yaml:
```diff
@@ -0,0 +1,451 @@
+    evaluationInterval: 15s
+    externalLabels:
+      cluster: cluster-name
+    remoteRead:
```
```suggestion
    remoteRead: {}
```

 Lets remove lookback. It will evaluate alerts with 1h delay.

f41gh7 reviewed (approved):
LGTM

f41gh7 commented:
Thanks for contribution!
